pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +381,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.forceAddDockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
I think `inlineDockerfile` tells *less* about the intent of the bool. For that to make sense, a reader needs to think about that coming from `oc new-build --dockerfile ...`, if that is what "inline" means in the context.

All the bool does for now, if you think about this PR + #6615 together, is to hint us into adding the Dockerfile contents, however and wherever it came from (because, sadly, it is quite hard to reason about it).

@bparees :+1: for a follow up.